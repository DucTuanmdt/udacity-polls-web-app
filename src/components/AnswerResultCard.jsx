import React from "react";

function AnswerResultCard({ option, totalAnswers, isSelected }) {
  return (
    <div
      className={`d-flex flex-column align-items-center gap-2 shadow px-3 py-4 rounded-2 ${
        isSelected ? "text-success border border-success" : "text-muted"
      }`}
    >
      <p className="h5">{option.text}</p>
      <p className="mb-0">
        <strong>Amount of votes:</strong> {option.votes.length}
      </p>
      <p className="mb-0">
        <strong> Percentage:</strong>{" "}
        {((option.votes.length / totalAnswers) * 100).toFixed(1)}%
      </p>
    </div>
  );
}

export default AnswerResultCard;
