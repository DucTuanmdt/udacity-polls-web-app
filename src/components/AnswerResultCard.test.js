import renderer from "react-test-renderer";
import AnswerResultCard from "./AnswerResultCard";

const defaultProps = {
  option: {
    votes: ["mtsamis", "sarahedo"],
    text: "Study React",
  },
  totalAnswers: 4,
  isSelected: false,
};

describe("Snapshot test for UserInfo.jsx", () => {
  test("should render unselected AnswerResultCard correctly", () => {
    const tree = renderer
      .create(<AnswerResultCard {...defaultProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("should render selected AnswerResultCard correctly", () => {
    const tree = renderer
      .create(<AnswerResultCard {...defaultProps} isSelected />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
