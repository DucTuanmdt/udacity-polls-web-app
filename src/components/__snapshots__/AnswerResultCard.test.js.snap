// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for UserInfo.jsx should render selected AnswerResultCard correctly 1`] = `
<div
  className="d-flex flex-column align-items-center gap-2 shadow px-3 py-4 rounded-2 text-success border border-success"
>
  <p
    className="h5"
  >
    Study React
  </p>
  <p
    className="mb-0"
  >
    <strong>
      Amount of votes:
    </strong>
     
    2
  </p>
  <p
    className="mb-0"
  >
    <strong>
       Percentage:
    </strong>
     
    50.0
    %
  </p>
</div>
`;

exports[`Snapshot test for UserInfo.jsx should render unselected AnswerResultCard correctly 1`] = `
<div
  className="d-flex flex-column align-items-center gap-2 shadow px-3 py-4 rounded-2 text-muted"
>
  <p
    className="h5"
  >
    Study React
  </p>
  <p
    className="mb-0"
  >
    <strong>
      Amount of votes:
    </strong>
     
    2
  </p>
  <p
    className="mb-0"
  >
    <strong>
       Percentage:
    </strong>
     
    50.0
    %
  </p>
</div>
`;
